packages/react/index.js

```diff
@@ -1,6 +1,36 @@
 'use strict';
 
+function testMinification() {
+  if (testMinification.name === 'testMinification') {
+    // We are not minified.
+    // This might be a Node environment where DCE is not expected anyway.
+    return;
+  }
+  if (process.env.NODE_ENV === 'production') {
+    if (process.env.NODE_ENV === 'development') {
+      throw new Error('This is unreachable');
+    }
+    try {
+      const source = testMinification.toString();
+      if (source.indexOf('toString') === -1) {
+        // We know for a fact the above line exists.
+        // Therefore the browser gave us invalid source.
+        return;
+      }
+      if (source.indexOf('unreachable') !== -1) {
+        // Dead code elimination would have stripped that branch
+        // because it is impossible to reach in production.
+        setTimeout(function() {
+          // Ensure it gets reported to production logging
+          throw new Error('React is running in production mode, but dead code elimination has not been applied.');
```

Comment:
This should probably include "Read how to correctly configure React for production: " and an fame link to "Use the Production Build" in "Optimizing Performance" doc.